doc/api/permissions.md

```diff
@@ -440,7 +446,153 @@ not adopt the origin of the `blob:` URL.
+  const result = internalModuleStat(filename);
+                 ^
+
+Error: Access to this API has been restricted
+    at stat (node:internal/modules/cjs/loader:171:18)
+    at Module._findPath (node:internal/modules/cjs/loader:627:16)
+    at resolveMainPath (node:internal/modules/run_main:19:25)
+    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:76:24)
+    at node:internal/main/run_main_module:23:47 {
+  code: 'ERR_ACCESS_DENIED',
+  permission: 'FileSystemRead'
+}
+```
+
+Allowing access to spawning a process and creating worker threads can be done
+using the [`--allow-child-process`][] and [`--allow-worker`][] respectively.
+
+#### Runtime API
+
+When enabling the Permission Model through the [`--experimental-permission`][]
+flag a new property `permission` is added to the `process` object.
+This property contains two functions:
+
+##### `permission.deny(scope [,parameters])`
+
+API call to deny permissions at runtime ([`permission.deny()`][])
+
+```js
+process.permission.deny('fs'); // Deny permissions to ALL fs operations
+
+// Deny permissions to ALL FileSystemWrite operations
+process.permission.deny('fs.write');
+// deny FileSystemWrite permissions to the protected-folder
+process.permission.deny('fs.write', ['/home/rafaelgss/protected-folder']);
+// Deny permissions to ALL FileSystemRead operations
+process.permission.deny('fs.read');
+// deny FileSystemRead permissions to the protected-folder
+process.permission.deny('fs.read', ['/home/rafaelgss/protected-folder']);
+```
+
+##### `permission.has(scope ,parameters)`
+
+API call to check permissions at runtime ([`permission.has()`][])
+
+```js
+process.permission.has('fs.write'); // true
+process.permission.has('fs.write', '/home/rafaelgss/protected-folder'); // true
+
+process.permission.deny('fs.write', '/home/rafaelgss/protected-folder');
+
+process.permission.has('fs.write'); // true
+process.permission.has('fs.write', '/home/rafaelgss/protected-folder'); // false
+```
+
+#### File System Permissions
+
+To allow access to the filesystem, use the [`--allow-fs-read`][] and
+[`--allow-fs-write`][] flags:
+
+```console
+$ node --experimental-permission --allow-fs-read=* --allow-fs-write=* index.js
+Hello world!
+(node:19836) ExperimentalWarning: Permission is an experimental feature
+(Use `node --trace-warnings ...` to show where the warning was created)
+```
+
+The valid arguments for both flags are:
+
+* `*` - To allow the all operations to given scope (read/write).
+* Paths delimited by comma (,) to manage reading/writing operations.
+
+Example:
+
+* `--allow-fs-read=*` - It will allow all `FileSystemRead` operations.
+* `--allow-fs-write=*` - It will allow all `FileSystemWrite` operations.
+* `--allow-fs-write=/tmp/` - It will allow `FileSystemWrite` access to the `/tmp/`
+  folder.
+* `--allow-fs-read=/tmp/,/home/.gitignore` - It allows `FileSystemRead` access
+  to the `/tmp/` folder **and** the `/home/.gitignore` path.
```

Comment:
What's the proper syntax for allowing write access to a single directory and its contents? I was hoping to try at least a few very simple cases but I must be misunderstanding something. I'd appreciate a hint.

<details>
<summary>What I tried so far</summary>
I built 75d3848be4995a9480266dbef066a8034283408d on Ubuntu 22.04 x64.

This works from within `/home/tniessen/dev/node`, where `b` is a directory in the current working directory:

```console
node --experimental-permission --allow-fs-write='*' -e "fs.writeFileSync('b/foo', 'test')"
```

```console
$ echo $(pwd) && test -d $(pwd)/b && echo "$(pwd)/b is a directory"
/home/tniessen/dev/node
/home/tniessen/dev/node/b is a directory
```

But none of these do, they all throw `ERR_ACCESS_DENIED`:

```console
node --experimental-permission --allow-fs-write='/home/tniessen/dev/node/b/*' -e "fs.writeFileSync('b/foo', 'test')"
```

```console
node --experimental-permission --allow-fs-write='/home/tniessen/dev/node/b/' -e "fs.writeFileSync('b/foo', 'test')"
```

```console
node --experimental-permission --allow-fs-write='/home/tniessen/dev/node/b' -e "fs.writeFileSync('b/foo', 'test')"
```

```console
node --experimental-permission --allow-fs-write='/home/tniessen/dev/node/b/foo' -e "fs.writeFileSync('b/foo', 'test')"
```

```console
node --experimental-permission --allow-fs-write='/home/tniessen/dev/node/' -e "fs.writeFileSync('b/foo', 'test')"
```

```console
node --experimental-permission --allow-fs-write="$(pwd)" -e "fs.writeFileSync('b/foo', 'test')"
```

```console
node --experimental-permission --allow-fs-write="$(pwd)/*" -e "fs.writeFileSync('b/foo', 'test')"
```
</details>

Reply:
I'll look into it

Reply:
It was a bug @tniessen. The tests didn't catch it because I forgot to include another path instead `*` in the cli tests. Fixed in b7a2f479ab.